addons/event_sale/wizard/event_configurator.py

```diff
@@ -11,3 +12,9 @@ class EventConfigurator(models.TransientModel):
     product_id = fields.Many2one('product.product', string="Product", readonly=True)
     event_id = fields.Many2one('event.event', string="Event")
     event_ticket_id = fields.Many2one('event.event.ticket', string="Event Ticket")
+
+    @api.constrains('event_id', 'event_ticket_id')
+    def check_event_id(self):
+        for record in self:
+            if record.event_id.id != record.event_ticket_id.event_id.id:
+                raise ValidationError(_('Invalid event / ticket choice(s)'))
```

Comment:
Could be great to add event name and ticket name directly in message (even if those are m2o and displayed behind the modal, it could be hidden with manual creates in the code, so always better to have complete error messages)